dvc/fs/fsspec_wrapper.py

```diff
@@ -37,10 +39,32 @@ def _entry_hook(self, entry):
         return entry
 
     def isdir(self, path_info):
-        return self.fs.isdir(self._with_bucket(path_info))
+        if not self.exists(path_info):
+            return False
+
+        # Directory in object storages are interpreted differently
+        # among different fsspec providers, so this logic is a temporary
+        # measure for us to adapt as of now. It checks whether it is a
+        # directory (as in a prefix with contents) or whether it is an empty
+        # file where it's name ends with a forward slash
+        entry = self.info(path_info)
+        return entry["type"] == "directory" or (
+            entry["size"] == 0
+            and entry["type"] == "file"
+            and entry["name"].endswith("/")
+        )
 
     def isfile(self, path_info):
-        return self.fs.isfile(self._with_bucket(path_info))
+        if not self.exists(path_info):
```

Comment:
@efiop forgot to mention this during our meeting, that the reason we have this guards instead of a try/except is that if I do this on isdir()
```
def isdir():
    try:
       ...
    except FileNotFoundError:
       return False
```

then `isfile()` is just going to negate the `False` to `True` even though the `False` is coming because the file doesn't exist (not because the file is not a directory). Since it is impossible to distinguish these operations, and for the sake of not repeating the same logic (if a file is empty and has trailing slash return False from isfile) I just used isdir() and negated the result but I have a pre-guard that ensures the file exists in the first place so that we won't return True when the file doesn't exist.